src/UI/Actions.hs

```diff
@@ -1486,16 +1516,14 @@ editAttachment s =
 -- is needed when editing parts during composition of an e-mail.
 --
 upsertPart ::
-     CharsetLookup
-  -> MIMEMessage
+  MIMEMessage
   -> L.List Name MIMEMessage
   -> L.List Name MIMEMessage
-upsertPart charsets newPart l =
+upsertPart newPart l =
   case L.listSelectedElement l of
     Nothing -> L.listInsert 0 newPart l
     Just (_, part) ->
-      if view (headers . contentDisposition . folded . filename charsets) part
-          == view (headers . contentDisposition . folded . filename charsets) newPart
+      if view headers part == view headers newPart
```

Comment:
So, now `upsertPart` will insert a new part of any headers changed.  I'm ok with that.  But it would help if the behaviour of the function were fully elaborated in the haddock.

Reply:
On further thought, although it's not something to tackle in this commit, I think we should take a different approach: we should store more state before we go to edit a (new or existing) part, so that we know what to do when editing returns, i.e.:

- if it is new, note that in the state
- if editing existing, store a "cursor" in the state, so that we know exactly what to update

That will allow e.g. editing of headers (which is how I like to write my mail - with all the headers there in the file being edited).

As I say, it is not for this commit.  I have always found `upsertPart` a rather confusing function.  And it turns out that the current approach is inflexible too.

Reply:
Maybe we can even pass a not-quite-saturated list update action along through the action, so we don't have to add anything in the application state.  Would be quite a refactor though, with updates to the `Action` data type and the `chain` behaviour.  Could be fun though :)

Reply:
I like the last approach, since you can implement what you mentioned in your second comment. I agree that the current approach is rather wonky :/ How much do you think does this idea relate to #294 ?

Reply:
I'll leave comments over in #33 294.